src/Dataflows/Transformations/Conversion.Converters.cs

```
@@ -0,0 +1,192 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 TTCO Holding Company, Inc. and Contributors
+// Licensed under the Apache License, Version 2.0
+// See https://opensource.org/licenses/Apache-2.0 or the LICENSE file in the repository root for the full text of the license.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+
+namespace Shipwright.Dataflows.Transformations
+{
+    public partial record Conversion
+    {
+        /// <summary>
+        /// Pre-defined conversion delegates.
+        /// </summary>
+
+        public static class Converters
+        {
+            /// <summary>
+            /// Creates a converter for integer values.
+            /// </summary>
+
+            public static ConverterDelegate Integer( NumberStyles styles = NumberStyles.Any, CultureInfo? cultureInfo = null ) => delegate ( object value, out object? result )
+            {
+                cultureInfo ??= CultureInfo.CurrentCulture;
+
+                if ( value is int converted || (value is string text && int.TryParse( text, styles, cultureInfo, out converted )) )
+                {
+                    result = converted;
+                    return true;
+                }
+
+                if ( value is IConvertible convertible )
+                {
+                    try
+                    {
+                        result = Convert.ToInt32( convertible );
+                        return true;
+                    }
+
+                    catch { }
```

If this is a case, similar to that in NET-1365, where you might be waiting to catch particular exceptions as the need is determined, wouldn't you want to use finally instead of an empty catch, which I think might just swallow all exceptions?

---

In this case, no. We're using an empty catch to swallow because we don't exactly care what the exception is (only that it happened); the method will return false indicating conversion failed.